rex-ast/src/expr.rs

```diff
@@ -77,16 +77,7 @@ pub enum Expr {
     Tuple(Id, Span, Vec<Expr>),             // (e1, e2, e3)
     List(Id, Span, Vec<Expr>),              // [e1, e2, e3]
     Dict(Id, Span, BTreeMap<String, Expr>), // {k1 = v1, k2 = v2}
-
-    // TODO(loong): in order to properly support ADT constructor function we
-    // need to support an expression like this:
-    //
-    // ```rex
-    // Named(Id, Span, String, Box<Type>), //  MyVariant1 {k1 = v1, k2 = v2}
-    // ```
-    //
-    // Note that it wouldn't be something that you can express in code, but
-    // rather something that can be returned by a function.
+    Named(Id, Span, String, Box<Expr>), //  MyVariant1 {k1 = v1, k2 = v2}
```

Comment:
variants don't have to have an inner type, so we shouldn't require an inner expression
```suggestion
    Named(Id, Span, String, Option<Box<Expr>>), //  MyVariant1 {k1 = v1, k2 = v2}
```